redshift/resource_redshift_group.go

```diff
@@ -213,6 +213,34 @@ func setGroupName(tx *sql.Tx, d *schema.ResourceData) error {
 	return nil
 }
 
+func checkIfUserExists(tx *sql.Tx, name interface{}) (bool, error) {
+
+	username, ok := name.(string)
+	if !ok {
+		return false, fmt.Errorf("cannot convert username to string")
+	}
+
+	sql := fmt.Sprintf("SELECT * FROM pg_user WHERE usename='%s';", username)
+
+	rows, err := tx.Query(sql)
+	if err != nil {
+		return false, err
+	}
+	defer rows.Close()
+
+	count := 0
+
+	for rows.Next() {
+		count += 1
+	}
+
+	if count > 0 {
+		return true, nil
+	}
+
+	return false, nil
```

Comment:
It is better to use a prepared statement here. Also, a `QueryRow` function will return an error of type `ErrNoRows` which will simplify the logic a bit:
```go
	var result int
	err := tx.QueryRow("SELECT 1 from pg_user_info WHERE usename=$1", name).Scan(&result)

	switch {
	case err == sql.ErrNoRows:
		return false, nil
	case err != nil:
		return false, fmt.Errorf("Error reading info about user: %s", err)
	}

	return true, nil
```

Reply:
good point.

btw error strings should not be capitalized in golang

Reply:
updated